core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -897,14 +902,55 @@ public void start() {
          displayName = "Stops cache."
    )
    public void stop() {
-      stop(null);
+      performImmediateShutdown();
    }
 
-   void stop(ClassLoader explicitClassLoader) {
-      if (log.isDebugEnabled()) log.debugf("Stopping cache %s on %s", getName(), getCacheManager().getAddress());
+   @Override
+   @ManagedOperation(
+         description = "Shuts down the cache",
+         displayName = "Shuts down the cache"
+   )
+   public void shutdown() {
+      if (log.isDebugEnabled())
+         log.debugf("Shutting down cache %s on %s", getName(), getCacheManager().getAddress());
+      if (!stopping) {
+         synchronized (this) {
+            // DCL to make sure that only one thread calls stop at one time,
+            // and any other calls by other threads are ignored.
+            if (!stopping && componentRegistry.getStatus() == ComponentStatus.RUNNING) {
+               stopping = true;
+               requestGracefulShutdown();
+            }
+         }
+      }
+   }
+
+   private void requestGracefulShutdown() {
```

Comment:
Shouldn't this be `requestClusterWideShutdown()`? `performGracefulShutdown()` does something quite different.